controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -510,6 +510,65 @@ class ControllerManager : public rclcpp::Node
   };
 
   SwitchParams switch_params_;
+
+  class ControllerThreadWrapper
+  {
+  public:
+    ControllerThreadWrapper(
+      std::shared_ptr<controller_interface::ControllerInterfaceBase> & controller,
+      int cm_update_rate)
+    : terminated_(false), controller_(controller), thread_{}, cm_update_rate_(cm_update_rate)
+    {
+    }
+
+    ControllerThreadWrapper(const ControllerThreadWrapper & t) = delete;
+    ControllerThreadWrapper(ControllerThreadWrapper && t) = default;
+    ~ControllerThreadWrapper()
+    {
+      terminated_.store(true, std::memory_order_seq_cst);
+      if (thread_.joinable())
+      {
+        thread_.join();
+      }
+    }
+
+    void start() { thread_ = std::thread(&ControllerThreadWrapper::call_controller_update, this); }
+
+    void call_controller_update()
+    {
+      rclcpp::Time previous_time = controller_->get_node()->now();
+
+      while (!terminated_.load(std::memory_order_relaxed))
+      {
+        auto const period = std::chrono::nanoseconds(1'000'000'000 / cm_update_rate_);
```

Comment:
What is the purpose of `cm_update_rate_`? We have now the same update rate as controller manager. I am not sure if this is the point of the excercise.

Reply:
If we don't have an update rate, then there's no way to prevent the async controller from using up all the CPU power, since it will call its update all the time. Would it make more sense to use the controller's own update rate, and the cm's if it isn't defined?